Formula/onlykey-agent.rb

```diff
@@ -0,0 +1,213 @@
+class OnlykeyAgent < Formula
+  include Language::Python::Virtualenv
+
+  desc "Middleware that lets you use OnlyKey as a hardware SSH/GPG device"
+  homepage "https://docs.crp.to/onlykey-agent.html"
+  url "https://files.pythonhosted.org/packages/02/c1/27c6cfbc5ee63fca91e37915d0182c0bfb988ca12362f01bcd5451e0ee10/onlykey-agent-1.1.14.tar.gz"
+  sha256 "096f20580ae112f57c1b9b279ed17728dc8e6f0fca301be586b9d976177f1523"
+  license "LGPL-3.0-only"
+
+  # rust according to https://cryptography.io/en/latest/installation/#rust
+  depends_on "rust" => :build
+
+  # libusb and gnupg according to https://docs.crp.to/onlykey-agent.html#installation
+  depends_on "gnupg"
+
+  depends_on "libusb"
+
+  # openssl according to https://cryptography.io/en/latest/installation/#supported-platforms:
+  depends_on "openssl@1.1"
+
+  depends_on "python@3.10"
+
+  # six come from brew create --python adding it as a resource
+  depends_on "six"
+
+  resource "aenum" do
+    url "https://files.pythonhosted.org/packages/63/6c/a71e18de7c651f384b328be6bccadbbd472aca62f547c1a307b9388d03ca/aenum-3.1.11.tar.gz"
+    sha256 "aed2c273547ae72a0d5ee869719c02a643da16bf507c80958faadc7e038e3f73"
+  end
+
+  resource "hidapi" do
+    url "https://files.pythonhosted.org/packages/ef/72/54273f701c737ae5f42d9c0adf641912d20eb955c75433f1093fa509bcc7/hidapi-0.12.0.post2.tar.gz"
+    sha256 "8ebb2117be8b27af5c780936030148e1971b6b7fda06e0581ff0bfb15e94ed76"
+  end
```

Comment:
I think you need a `depends_on "hidapi"` for this to be useful?

Reply:
I'm happy to add it, but I don't know if this is necessary or not. FWIW I noticed it isn't in https://github.com/Homebrew/homebrew-core/blob/master/Formula/trezor-agent.rb which I know onlykey-agent is based on. @onlykey do you have advice here?

Reply:
PyPI (cython-)`hidapi` vendors a copy of `hidapi`.

It takes some extra work to override with system copy based on my previous PR for `liquidctl`. This is due to the extra command line arg (`--with-system-hidapi`) needed to force this:
https://github.com/Homebrew/homebrew-core/blob/693b3239cda90f7dbf0f42cd470333bb96a370e3/Formula/liquidctl.rb#L55-L60

Reply:
Personally I prefer letting [`cython-hidapi`](https://github.com/trezor/cython-hidapi/blob/master/setup.py) vendor hidapi over monkey patching it, but maybe I'm missing something?

If using the homebrew hidapi is important enough maybe coming up with a patch to cython-hidapi to allow passing in the system hidapi include prefix better?

Reply:
> Personally I prefer letting [`cython-hidapi`](https://github.com/trezor/cython-hidapi/blob/master/setup.py) vendor hidapi over monkey patching it, but maybe I'm missing something?

See https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae.

> If using the homebrew hidapi is important enough maybe coming up with a patch to cython-hidapi to allow passing in the system hidapi include prefix better?

If we can upstream it, that would be nice.

Reply:
@prusnak Would you be open to a setup arg similar to `--with-system-hidapi` that would allow providing an include prefix to use system libraries rather than the vendored `hidapi`  (more context in the discussion above)?

Reply:
Isn't --with-system-hidapi exactly that option?

Reply:
FYI: This is resolved with the monkey patch in a8419fe04d18e6d22bf537a9f9f05a4373d309c6 . I wanted to unblock this formula for now and will consider the upstream to cython-hidapi after I get this merged.